Lumio updates

hdrei opened the pull request:
**DO NOT MERGE YET**

This is a draft PR and the pages might still contain some comments that are not meant to be published. If you have time to give feedback (here or for example in RC) or if you wish to contribute directly, now it starts to be good timing for that. There's at least something (that should be commented out) that should be checked, when LUMI-O is available again. 

I've taken some ideas from the Allas docs https://docs.csc.fi/data/Allas/ and at least it would be good to check more closely what Kurt has been teaching (and the material) in the latest LUMI intro training. 

Notice that some unfinished/planned chapters are commented out, and only visible in the markdown text. Also I'm still wondering how/where some contents should be placed. More use case examples could be added later. 
I checked an option to add content also to the heading pages (make them clickable with content on them). There are "commented out additions" in requirements.txt and mkdocs.yml that would allow this addition. I did not add it as an active change now, but I've tested that it works (with a local build) if we want to add this feature later. 

I've kept the existing url's of the main pages in the table of contents of the existing LUMI-O docs. Some existing links that point to a heading on a page might not work anymore with this version. 

https://lumi-supercomputer-docs-preview.rahtiapp.fi/origin/lumio-updates/storage/lumio/

hdrei committed:
Draft-1 of modifications to LUMI-O docs

hdrei committed:
Merge branch 'production' into lumio-updates

hdrei committed:
additions&modifications

hdrei committed:
More updates to LUMI-O docs 2

hdrei committed:
Resolving merge conflict in mkdocs.yml by keeping all new additions

hdrei commented:
Also there were some changes that I've made that I've checked earlier that works with LUMI-O, but I'd like to confirm (from someone who actually knows) if these are the best practices and I didn't miss something relevant with the guidelines

mszpindler commented:
@hdrei this is a solid piece of work! The only concern I have is if it is not too fragmented (across too many individual pages) now.

hdrei committed:
Applying feedback part1

hdrei committed:
Merge branch 'production' into lumio-updates

hdrei committed:
small update

hdrei committed:
Applying more changes based on the feedback

hdrei committed:
Merge branch 'production' into lumio-updates

hdrei committed:
more changes

mszpindler commented on docs/storage/lumio/index.md:
```diff
@@ -4,224 +4,91 @@
+    - Usage with client tools like rclone and s3cmd, or via LUMI web interface
+    - Possible to set granular access rights to data (e.g. for other projects or users from other projects) 
+    - Possible to set a temporary access link to otherwise private data
+    - Possible to share pubic data across the internet (for the duration of your project)
```
```suggestion
    - Possible to share public data across the internet (for the duration of your project)
```

mszpindler commented on docs/storage/lumio/index.md:
```diff
@@ -4,224 +4,91 @@
 
-Within your object storage project space, you can create buckets. These buckets
-will store objects with metadata associated with these objects.
+LUMI-O is a S3 compatible object storage. If one has used Amazon AWS S3, many functionalities are the same with LUMI-O, but one should [be aware of the differences](#lumi-o-vs-amazon-s3).
```
```suggestion
LUMI-O is object storage service accessible through a web interface and S3 API. 
```

mszpindler commented on docs/storage/lumio/index.md:
```diff
@@ -4,224 +4,91 @@
-Within your object storage project space, you can create buckets. These buckets
-will store objects with metadata associated with these objects.
+LUMI-O is a S3 compatible object storage. If one has used Amazon AWS S3, many functionalities are the same with LUMI-O, but one should [be aware of the differences](#lumi-o-vs-amazon-s3).
+LUMI-O is based on completely different technology than the Lustre parallel filesystem (LUMI-P and LUMI-F). Also, unlike the Lustre parallel file system, LUMI-O is a separate system from LUMI. 
```
```suggestion
LUMI-O is based on completely different technology than the Lustre parallel filesystem (LUMI-P and LUMI-F). Also, unlike the Lustre parallel file system, LUMI-O is a separate service from LUMI compute partitions. 
```

mszpindler commented on docs/storage/lumio/index.md:
```diff
@@ -4,224 +4,91 @@
+
+All LUMI projects have LUMI-O available by default.
+
+To access LUMI-O, you will need to [generate authentication credentials](./auth-lumidata-eu.md) that are specific to accessing LUMI-O, and not the same thing as the SSH-keys to access LUMI.
```
```suggestion
To access LUMI-O, you will need to [generate access token](./auth-lumidata-eu.md). This token is exclusive for your project. There are no other credentials allowing access to LUMI-O bucket. 
```

mszpindler commented on docs/storage/lumio/index.md:
```diff
@@ -4,224 +4,91 @@
+
+## Structure of object storage
+
+Structure of an object storage differs a bit from a normal file system structure. Instead of a filesystem structure with directories and files, the data is organized in a structure with *buckets* that can contain *objects*. 
```
```suggestion
Structure of an object storage is different from a normal file system structure. Instead of directories and files, the data is organized in a flat structure with *buckets* that contain *objects*. 
```

mszpindler commented on docs/storage/lumio/auth-lumidata-eu.md:
```diff
@@ -105,18 +128,15 @@ provides the necessary "Access key" and "Secret key". You can also see the key
 !!! info
 	When you initially create the key, the initial allowed maximum lifetime is 168h = 7 days.
 	This can then be extended by one hour for every hour from key creation,
     i.e., you are always able to extend the duration of the key to now + 7 days.
```
```suggestion
    i.e., you are always able to extend the duration of the key to now + 7 days. Default lifetime of the key is 24 hours.
```

mszpindler commented on docs/storage/lumio/advanced.md:
```diff
@@ -1,138 +1,318 @@
+# Sharing access to data
 
-## Introduction
+Sharing data, e.g. datasets or research results, is (in principle) easy in the object storage. One should pay attention to understand what they are doing, though, especially before changing access rights to the data (with ACLs or Policies), or to not unintendedly publish something that shouldn't be publicly accessible.
```
I would try to make this and next paragraphs more compact, there is too much general text here. Emphasise more on different sharing options.

hdrei replied:
I tried to reduce the amount of "introductory" text on this page. Deleted some (that was maybe somewhat repetition) and moved some..

mszpindler commented on docs/storage/lumio/auth-lumidata-eu.md:
```diff
@@ -131,3 +151,93 @@ can be created in its place.
 </figure>
 
+
+## Connecting
```
I would not use term "connecting". Also I would keep this page for authorization only (tokens) and move "connecting" part to the next page "transferring and managing data".

hdrei replied:
Tried to get rid of the word "connecting" (in the sense of "connecting to LUMI-O"), I agree that it doesn't make sense. Replaced it mostly with word "accessing". 
I think that the connecting/accessing part suits well on the same page as information about the access credentials. I see though that what it comes to creating rclone and s3cmd config files (~ `module load lumio`, `lumio-conf`), it would also suit on the 'Managing data' page. But if we would like to list all the options how to "Access LUMI-O" including the web interface, I would prefer placing it on the same page with credentials and call the page "Accessing LUMI-O". 
I modified the content a bit. What do you think about it now?

hdrei replied:
There's also the advanced part about 'Credentials and configuration' that is now on the 'Accessing LUMI-O' page. I see that this could be also on the 'Managing data' page, but I also kind of like to keep the "authentication and credentials" out of the "basic" usage what it comes to transferring data to/from LUMI-O.
I'm not completely certain of this opinion, though. If you still strongly feel you would like to move all the "access and config files" content to the 'Managing data' page, I can try to figure out how to put this

hdrei replied:
Since no further comments, marking this as resolved

mszpindler commented on docs/storage/lumio/clients-general.md:
```diff
@@ -0,0 +1,424 @@
+# Transferring and managing data
```
I would use "Managing data" as the name for this page. It makes menu navigation easier.

mszpindler commented on docs/storage/lumio/advanced.md:
```diff
@@ -1,138 +1,318 @@
-# Advanced usage of LUMI-O  
+# Sharing access to data
```
I would use "Sharing data" as the name for this page. It makes menu navigation easier.

mszpindler commented on docs/storage/lumio/connect-local.md:
```diff
@@ -0,0 +1,34 @@
+# Using LUMI-O without connecting to LUMI
```
I would combine this page with "simple upload and download". It is basically the same, requires extra steps for configuration lumi-o modules can do on LUMI.

hdrei replied:
I disagree with this one, for several reasons. The 'Access without connecting to LUMI' is now a bit short, but I was thinking to possibly add some guidance for Windows users also (at a later point after this PR). Also I don't think this is part of the most simple usage for people who are new to the whole concept of object storages, and possibly also with rclone. Also personally I like it that it is a separate use case example and well visible as a topic that "you can actually access LUMI-O without connecting to LUMI", since this is not self evident at all for a "basic user"

hdrei replied:
Since no further comments, marking this as resolved

mszpindler commented on docs/storage/lumio/between-object-storages.md:
```diff
@@ -0,0 +1,137 @@
+# Copying files directly between object storages
```
"object storages" sounds weird to me. Could we rephrase this, "between object storage endpoints" for instance?

hdrei committed:
Update docs/storage/lumio/index.md

Co-authored-by: Maciej Szpindler <56594900+mszpindler@users.noreply.github.com>

hdrei committed:
Update docs/storage/lumio/index.md

Co-authored-by: Maciej Szpindler <56594900+mszpindler@users.noreply.github.com>

hdrei committed:
Update docs/storage/lumio/index.md

To access LUMI-O, you will need to [generate access tokens](./auth-lumidata-eu.md). These tokens are personal and exclusive for a project. There are no other credentials allowing access to LUMI-O buckets.

Co-authored-by: Maciej Szpindler <56594900+mszpindler@users.noreply.github.com>

hdrei committed:
Update docs/storage/lumio/index.md

Co-authored-by: Maciej Szpindler <56594900+mszpindler@users.noreply.github.com>

hdrei committed:
Update docs/storage/lumio/index.md

Co-authored-by: Maciej Szpindler <56594900+mszpindler@users.noreply.github.com>

hdrei committed:
Update docs/storage/lumio/auth-lumidata-eu.md

Co-authored-by: Maciej Szpindler <56594900+mszpindler@users.noreply.github.com>

hdrei committed:
changes based on feedback

hdrei committed:
Merge branch 'production' into lumio-updates

mszpindler commented on docs/storage/lumio/connect-local.md:
```diff
@@ -0,0 +1,36 @@
+# Using LUMI-O without connecting to LUMI
```
```suggestion
# Using LUMI-O from your local computer
```

mszpindler commented on docs/storage/lumio/connect-local.md:
```diff
@@ -0,0 +1,36 @@
+
+
+
+This example is with `rclone`. The same works similarily with e.g `s3cmd` and other clients. (For s3cmd the default path under home directory is `~/.s3cfg`.)
```
```suggestion
This example uses `rclone`. The same works similarly with e.g. `s3cmd` and other clients. (for the s3cmd the default path under home directory is `~/.s3cfg`.)
```

mszpindler commented on docs/storage/lumio/connect-local.md:
```diff
@@ -0,0 +1,36 @@
+
+This example is with `rclone`. The same works similarily with e.g `s3cmd` and other clients. (For s3cmd the default path under home directory is `~/.s3cfg`.)
+
+By setting the file `rclone.conf` in the default path `~/.config/rclone/rclone.conf` (e.g. in your own laptop) with the correct configuration setting and valid access keys, it is possible to use rclone to access LUMI-O from anywhere across internet, e.g. from your own laptop. Connection to LUMI is not needed. 
```
```suggestion
By setting the file `rclone.conf` (default config path is `~/.config/rclone/rclone.conf`) on your local system with the correct configuration setting and valid access key, rclone opens LUMI-O session from anywhere across internet, e.g. from your own laptop. Connection to LUMI is not needed. 
```

mszpindler commented on docs/storage/lumio/connect-local.md:
```diff
@@ -0,0 +1,36 @@
+
+By setting the file `rclone.conf` in the default path `~/.config/rclone/rclone.conf` (e.g. in your own laptop) with the correct configuration setting and valid access keys, it is possible to use rclone to access LUMI-O from anywhere across internet, e.g. from your own laptop. Connection to LUMI is not needed. 
+
+Notice that you should have rclone installed for this example to work. If you don't have rclone installed, follow the directions e.g. [on rclone website](https://rclone.org/install/) to install rclone on your laptop.
```
```suggestion
Note you should have rclone installed locally for this example to work. Follow the directions e.g. [on rclone website](https://rclone.org/install/) to install rclone on your local computer.
```

mszpindler commented on docs/storage/lumio/connect-local.md:
```diff
@@ -0,0 +1,36 @@
+
+**3)** Select `rclone` from the Configuration templates and click 'Generate'
+
+**4)** Copy the created text block to a file (on your laptop) that you name 'rclone.conf' (or add it to an existing rclone.conf file). It's simplest to place it in the default path for rclone `~/.config/rclone/rclone.conf` 
```
```suggestion
**4)** Copy the created text block to the local 'rclone.conf' file (create the file or add it to an existing file). Default path for rclone config is `~/.config/rclone/rclone.conf` 
```

mszpindler commented on docs/storage/lumio/connect-local.md:
````diff
@@ -0,0 +1,36 @@
+```
+rclone lsd lumi-46500XXXX-private:
+```
+to see the buckets of your project 46500XXXX in LUMI-O. (_Replace 46500XXXX with your actual LUMI project number_.)
````
```suggestion
to see existing buckets of your project 46500XXXX in LUMI-O. (_Replace `46500XXXX` with your actual LUMI project ID.)
```

hdrei committed:
Update docs/storage/lumio/connect-local.md

Co-authored-by: Maciej Szpindler <56594900+mszpindler@users.noreply.github.com>

hdrei committed:
Update docs/storage/lumio/connect-local.md

Co-authored-by: Maciej Szpindler <56594900+mszpindler@users.noreply.github.com>

hdrei committed:
Update docs/storage/lumio/connect-local.md

Co-authored-by: Maciej Szpindler <56594900+mszpindler@users.noreply.github.com>

hdrei committed:
Update docs/storage/lumio/connect-local.md

Note that you should have rclone installed locally for this example to work. Follow the directions e.g. [on rclone website](https://rclone.org/install/) to install rclone on your local computer.

Co-authored-by: Maciej Szpindler <56594900+mszpindler@users.noreply.github.com>

hdrei committed:
Update docs/storage/lumio/connect-local.md

Co-authored-by: Maciej Szpindler <56594900+mszpindler@users.noreply.github.com>

hdrei committed:
Update docs/storage/lumio/connect-local.md

Co-authored-by: Maciej Szpindler <56594900+mszpindler@users.noreply.github.com>

hdrei committed:
some small additional fixes

TheBlackKoala commented on docs/storage/lumio/advanced.md:
Should probably be renamed "sharing" or similar

hdrei replied:
Other people wanted to keep the current urls working/existing, so did not change this now. Although I agree that it's not anymore the most describing name for the page

TheBlackKoala commented on docs/storage/lumio/allas-lumi.md:
```diff
@@ -0,0 +1,57 @@
+
+## Accessing Allas from LUMI
+
+...
```
This part is missing

hdrei replied:
This page is not included in this PR. I think the guidance about Allas should be in CSC docs. This instruction can be added later, if it's seen the best then. Will check with CSC docs about adding this example there. The "Accessing Allas from LUMI" is already actually documented in CSC docs

TheBlackKoala commented on docs/storage/lumio/allas-lumi.md:
````diff
@@ -0,0 +1,57 @@
+allas-conf -k project_<CSC-project-number>
+```
+
+The option `-k` should keep the connection open for 8 hours (at least on Puhti and Mahti, haven't checked this on LUMI yet, but I assume it's the same) without prompting the user.
````
maybe verify or leave as issue on git - comment should not be in official documentation (maybe something like "unverified on LUMI" would mark it as unknown for lumi

hdrei replied:
Yes this page is unfinished, and not included in this PR (explained in the comment below)

TheBlackKoala commented on docs/storage/lumio/between-object-storages.md:
````diff
@@ -0,0 +1,140 @@
+### General example
+
+```
+[lumi-46YXXXXXX-private]
````
the mix of Y and X can be confusing - especially in the between two lumi-o projects

TheBlackKoala commented on docs/storage/lumio/check-quota.md:
```diff
@@ -0,0 +1,60 @@
+- One project can have up to 1000 buckets
+- One bucket can have up to 500 000 objects
+
+If you need more storage space in LUMI-O, please contact the LUMI helpdesk. 
```
maybe add a link to the LUST-helpdesk?

hdrei replied:
Again "ghost pages", sorry... The link is included where this is actually presented on the other pages

TheBlackKoala commented on docs/storage/lumio/check-quota.md:
```diff
@@ -0,0 +1,60 @@
+
+### rclone
+
+Replace 46YXXXXXX with your LUMI project number.
```
again YXXX?

TheBlackKoala commented on docs/storage/lumio/clients-general.md:
```diff
@@ -0,0 +1,411 @@
+| Create bucket *mybuck*                     | `rclone mkdir lumi-46YXXXXXX-private:mybuck`         |
+| List objects in bucket *mybuck*            | `rclone ls lumi-46YXXXXXX-private:mybuck/`           |
+| Upload file *file1* to bucket *mybuck*     | `rclone copy file1 lumi-46YXXXXXX-private:mybuck/`   |
+| Download file *file1* from bucket *mybuck* | `rclone copy lumi-46YXXXXXX-private:mybuck/file1 .`  |
```
Again YXXX - but very nice table!

hdrei replied:
Thanks, it's originally by Orian, just made some modifications for page structure and so

TheBlackKoala commented on docs/storage/lumio/clients-general.md:
```diff
@@ -0,0 +1,411 @@
+-->
+
+
+## Checking your utilized LUMI-O quota
```
This sections looks like a copy of check-quota.md?

TheBlackKoala replied:
And Also contains the YXXX

hdrei replied:
The check-quota.md is not included to this PR. I should actually remove that page, there are a few unnecessary "ghost pages" now...

TheBlackKoala commented on docs/storage/lumio/error-messages.md:
```diff
@@ -9,7 +9,8 @@ EntityTooLarge
 ### HTTP status code 403
 
-You have reached a quota limit. Contact your local Resource Allocator and request a quota increase. Please specify the project name, bucket name and size of the file in the request.
+You have reached a quota limit. If you need more quota in LUMI-O, please contact LUMI helpdesk.  
```
Maybe a link here?

hdrei replied:
Again "ghost page" thing

TheBlackKoala commented on docs/storage/lumio/lumio-batchjobs.md:
```diff
@@ -0,0 +1,31 @@
+#Batch job things
+
+#do something e.g. with rclone, e.g. download data:
+rclone copy lumi-46YXXXXXX-private:mybucket/mydata.txt ./
```
YXXX

TheBlackKoala commented on docs/storage/lumio/upload-download.md:
````diff
@@ -0,0 +1,77 @@
+Now you are able to use e.g. `rclone` commands to interact with LUMI-O. To list the buckets that have been created in LUMI-O for your project, run the following command:
+
+```
+rclone lsd lumi-46YXXXXXX-private:   #Replace 46YXXXXXX with your LUMI project number
````
This page also contains a lot of YXXX

TheBlackKoala reviewed (approved):
A few comments on a few things that could be changed to increase ease of reading etc. Looks good and a very welcome addition to the documentation!

mszpindler commented:
@hdrei please remember to remove all html commented content before merging.

hdrei committed:
small change to project id prsentations based on feedback

hdrei committed:
removing html comments

hdrei committed:
final edit: getting rid of some ghost-pages
